src/lib/media/Media.ts

```diff
@@ -9,7 +9,7 @@ export abstract class Media {
   /**
    * Metadata about the Media object
    */
-  _meta: ImageMeta | VideoMeta;
+  _meta: ImageMeta | VideoMeta | ContentReferenceMeta;
```

Comment:
I'm not sure ContentReference is a type of media. This was the base class for image and video links because both types of objects share a set of common fields (endpoint, name etc).

Since a content reference does not share those fields it may be more logical to have be its own thing. I would be tempted to put it under /content/model as ContentReference and make it independent.

Reply:
I was thinking along the same lines, shall review and move things around a bit, is there any core functionally the content reference class should have?

Reply:
Sounds good to me, simple to start seems good.